utils/mil_tools/nodes/label_me_bag.py

```diff
@@ -0,0 +1,268 @@
+          for segment in bag['segments']:
+              verify_attr(segment, 'start', 'segment', throw=False)
+              verify_attr(segment, 'stop', 'segment', throw=False)
+              verify_attr(segment, 'freq', 'segment', throw=False)
+              verify_attr(segment, 'topic', 'segment')
+              verify_attr(segment, 'name', 'segment')
+              self._print("\t\tFound segment '{}' from topic '{}' from {} to {} every {}",
+                          segment['name'], segment['topic'],
+                          segment['start'] if segment.has_key('start') else 'start',
+                          segment['stop'] if segment.has_key('stop') else 'stop',
+                          str(segment['freq']) + ' seconds' if segment.has_key('freq') else 'frame')
+
+  def read_bags(self):
+      """
+      Crawls through all bags in config YAML, placing images from the specified
+      topics and the specified frequency into corosponding folders within
+      Images of the labelme instalation.
+      
+      These folders will be created as follows if not already:
+      LABELME_DIR/Images/SEGEMENT_NAME/TOPIC_NAME
+      where TOPIC_NAME replaces all / with @ for compatibility with filesystems.
+      """
+      for bag in self.config['bags']:
+          self._print("Opening {}", bag['file'])
+          self._read_bag(bag['file'], bag['segments'])
+
+  def extract_labels(self):
+      """
+      Finds labeled XML files in LabelMe directories determined from the
+      config file, placing parsed annotation data as ros messages in a new bag
+      """
+      self._print("Extracting labels for all bags in config")
+      for bag in self.config['bags']:
+          self._print("\tExtracting LabelMe annotations for {}".format(bag['file']))
+          # TODO: give output bag a default name like INFILE_labeled.bag
+          outfile = bag['labeled_file']
+          self._extract_labels_bag(bag, outfile)
+
+  def print_report(self):
+      """
+      Compares how many images are present in each LabelMe directory affected
+      by the config YAML to the number of annotated XML files produced by
+      LabelMe in each of these directories.
+      
+      Prints out these counts in percentages by segement, bag, and overall
+      """
+      self._print("Generating completion report for all bags in config")
+      total_xml_count = 0
+      total_img_count = 0
+      for bag in self.config['bags']:
+          self._print("\tGenerating completion report for {}", bag['file'])
+          x, i = self._print_bag_report(bag)
+          total_xml_count += x
+          total_img_count += i
+      if total_img_count == 0:
+          print "{}/{} TOTAL images labeled (0%)".format(total_xml_count, total_img_count)
+      else:
+          print "{}/{} TOTAL images labeled ({:.1%})".format(total_xml_count, total_img_count, total_xml_count/total_img_count)
+
+  def _read_bag(self, bagfile, segments):
```

Comment:
Please add a documentation string to this function. Comments would also be nice if you have time.